surface/CMakeLists.txt

```diff
@@ -1,6 +1,10 @@
 set(SUBSYS_NAME surface)
 set(SUBSYS_DESC "Point cloud surface library")
-set(SUBSYS_DEPS common search kdtree octree)
+if(BUILD_surface_afront_debug)
+  set(SUBSYS_DEPS common search kdtree octree geometry visualization)
+else(BUILD_surface_afront_debug)
+  set(SUBSYS_DEPS common search kdtree octree geometry)
+endif(BUILD_surface_afront_debug)
```

Comment:
This is the perfect file to define your `BUILD_surface_afront_debug` option. But this block needs to move below line 15, since if the surface module is not build, this option should not pop-up.

You're adding a dependency to `geometry` which in turn only depends of `common`. `geometry` has a number of meshing primitives so I would say that functionally this dependency makes sense. To be checked is exactly what you use from there.

Your conditional branch should not set SUBSYS_DEPS directly, it should just append the visualization dependency to it. I want the module mandatory dependencies to appear on top first, isolated from whatever options the user decides later down the road, in case the `surface`module is built